File: miki/2.go_basics/even_ended/even_ended.js
console.time("start")

const min = 1000
const max = 9999

let count = 0

for (let i = min; i <= max; i++) {
  for (let j = i; j <= max; j++) {
    const n = i * j
    const s = String(n)
    if (s[0] === s[s.length-1]) {
      count++
    }
  }
}

console.log("found", count, "even-ended numbers")
console.timeEnd("start")

File: miki/2.go_basics/even_ended/even_ended.go
package main

import (
	"fmt"
	"strconv"
	"time"
)

func main() {
	start := time.Now()
	var count int

	min := 1000
	max := 9999

	for i := min; i <= max; i++ {
		for j := i; j <= max; j++ {
			n := i * j
			s := strconv.Itoa(n)
			if s[0] == s[len(s)-1] {
				count++
			}

		}
	}

	d := time.Since(start)
	fmt.Println("found", count, "even-ended numbers")
	fmt.Println("time took", d.Seconds(), "secs.")
}
